components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -143,12 +146,12 @@ void BraveReferralsService::Start() {
       this, &BraveReferralsService::OnFetchReferralHeadersTimerFired);
   DCHECK(fetch_referral_headers_timer_->IsRunning());
 
-  // On first run, read the promo code from user-data-dir and
-  // initialize the referral.
-  bool checked_for_promo_code_file =
-      pref_service_->GetBoolean(kReferralCheckedForPromoCodeFile);
+  // Read the promo code from user-data-dir and initialize the referral,
+  // retrying if necessary.
+  bool reported_installation =
+      pref_service_->GetBoolean(kReferralReportedInstall);
   std::string download_id = pref_service_->GetString(kReferralDownloadID);
-  if (!checked_for_promo_code_file && download_id.empty())
+  if (!reported_installation && download_id.empty())
```

Comment:
Just curious.. With this change, default code will be set again. Is this ok?

Reply:
Good point. No we don't want that. I went ahead and reintroduced `kReferralCheckForPromoCodeFile`. It just registers it to false and never modifies the values. For the existing users that have already set this preference, it will skip initialization. For everyone else it stays false forever, and we can phase this out on later releases. 

https://github.com/brave/brave-core/blob/535d5d301d04d6cb5a765f34de258df26c48c11b/components/brave_referrals/browser/brave_referrals_service.cc#L157-L162